pkg/asset/agent/image/unconfigured_ignition.go

```diff
@@ -164,42 +164,39 @@ func (a *UnconfiguredIgnition) Generate(_ context.Context, dependencies asset.Pa
+	rendezvousHostTemplateData := getRendezvousHostEnvTemplate(agentTemplateData, agentWorkflow.Workflow)
+	rendezvousHostTemplateFile := ignition.FileFromString(fmt.Sprintf("%s.template", rendezvousHostEnvPath), "root", 0644, rendezvousHostTemplateData)
+	config.Storage.Files = append(config.Storage.Files, rendezvousHostTemplateFile)
 
-	case workflow.AgentWorkflowTypeInstallInteractiveDisconnected:
-		// Add the rendezvous host file. Agent TUI will interact with that file in case
-		// the rendezvous IP wasn't previously configured, by managing it as a template file.
-		rendezvousIP := "{{.RendezvousIP}}"
-		if agentConfig.Config != nil {
-			rendezvousIP = agentConfig.Config.RendezvousIP
-		}
-		// Avoids escaping in case the template parameter was used.
-		rendezvousHostData, err := url.QueryUnescape(getRendezvousHostEnv("http", rendezvousIP, "", "", agentWorkflow.Workflow))
+	rendezvousIP, err := RetrieveRendezvousIP(agentConfig.Config, nil, nmStateConfigs.Config)
+	if err == nil {
+		rendezvousHostData, err := getRendezvousHostEnvFromTemplate(rendezvousHostTemplateData, rendezvousIP)
 		if err != nil {
 			return err
 		}
 		rendezvousHostFile := ignition.FileFromString(rendezvousHostEnvPath, "root", 0644, rendezvousHostData)
 		config.Storage.Files = append(config.Storage.Files, rendezvousHostFile)
+	}
 
-		// Explicitly disable the load-config-iso service, not required in the current flow
-		// (even though disabled by default, the udev rule may require it).
-		config.Storage.Files = append(config.Storage.Files, ignition.FileFromString("/etc/assisted/no-config-image", "root", 0644, ""))
-
-		// Enable the UI service.
-		enabledServices = append(enabledServices, "agent-start-ui.service")
-		interactiveUIFile := ignition.FileFromString("/etc/assisted/interactive-ui", "root", 0644, "")
-		config.Storage.Files = append(config.Storage.Files, interactiveUIFile)
+	switch agentWorkflow.Workflow {
+	case workflow.AgentWorkflowTypeInstall:
+		agentTemplateData.ConfigImageFiles = strings.Join(GetConfigImageFiles(), ",")
 
-		// Enable the agent-extract-tui service
-		enabledServices = append(enabledServices, "agent-extract-tui.service")
+		// Enable the agent-check-config-image.service for the current workflow.
+		enabledServices = append(enabledServices, "agent-check-config-image.service")
 
-		// Let's disable the assisted-service authentication.
-		agentTemplateData.AuthType = "none"
+		// To Do:
+		if sentinelfiel is present or another way to identify we want to invoke AgentWorkflowTypeInstallInteractiveDisconnected workflow {
+			// Add the rendezvous host file. Agent TUI will interact with that file in case
+			// the rendezvous IP wasn't previously configured, by managing it as a template file.
+			if rendezvousIP == "" {
+				rendezvousHostFile := ignition.FileFromString(rendezvousHostEnvPath, "root", 0644, rendezvousHostTemplateData)
+				config.Storage.Files = append(config.Storage.Files, rendezvousHostFile)
+			}
```

Comment:
this is coming from https://github.com/openshift/installer/pull/9941/files#diff-023fc980d398bf6069f7ad7406b50b1c250b46a1ad788bef31febee716f80bd4R191-R193